tmt/utils.py

```diff
@@ -2632,18 +2634,47 @@ def remove_color(text: str) -> str:
     return re.sub(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])', '', text)
 
 
-def default_branch(repository: Path, remote: str = 'origin') -> Optional[str]:
+def default_branch(repository: Path, logger: tmt.log.Logger,
+                   remote: str = 'origin') -> Optional[str]:
     """ Detect default branch from given local git repository """
+    # Make sure '.git' (which is a file or a directory) is present
+    dot_git = repository / '.git'
+    if not dot_git.exists():
+        return None
+
+    if not dot_git.is_file() and not dot_git.is_dir():
+        return None
+
+    work_repo = repository
+    if dot_git.is_file():
+        # Get the parent path of the worktree and the .git file looks like:
+        #     gitdir: /tmp/foo/REPRO/.git/worktrees/TREE
+        command = Command("git", "rev-parse", "--path-format=absolute", "--git-common-dir")
+        try:
+            result = run_command(command=command, cwd=Path(repository), logger=logger)
+        except BaseException:
+            return None
+        if result.stdout is None:
+            return None
+        git_parent_path = result.stdout.strip().replace('/.git', '')
+
+        # Should get the default branch of its parent
+        work_repo = Path(git_parent_path)
+
     # Make sure the '.git/refs/remotes/{remote}' directory is present
-    git_remotes_dir = repository / f'.git/refs/remotes/{remote}'
+    git_remotes_dir = work_repo / f'.git/refs/remotes/{remote}'
     if not git_remotes_dir.exists():
         return None
 
-    head = repository / f'.git/refs/remotes/{remote}/HEAD'
+    head = git_remotes_dir / 'HEAD'
     # Make sure the HEAD reference is available
     if not head.exists():
-        subprocess.run(
-            f'git remote set-head {remote} --auto'.split(), cwd=repository)
+        command = Command('git', 'remote', 'set-head', f'{remote}', '--auto')
+        try:
+            run_command(command=command, cwd=Path(work_repo), logger=logger)
+        except BaseException:
+            return None
```

Comment:
The same set of comments as above :)

1. `work_repo` seems to be `Path` already (`work_repo = repository`, `work_repo = Path(git_parent_path)`).
2. catching `BaseException` is probably incorrect, `Exception` or `RunError` would be better.
3. `command` as a one-shot variable.

Reply:
Fixed in 27e8add5.